Makefile.kokkos

```diff
@@ -24,33 +24,36 @@ KOKKOS_OPTIONS ?= ""
 # Options: force_uvm,use_ldg,rdc,enable_lambda
 KOKKOS_CUDA_OPTIONS ?= ""
 
+#TODO - I think this will work:
```

Comment:
Okay, we need some comment here and also I'd like to have a systematic naming of functions which get added and are not native to cmake. Lets for example do the following convention:

Variable all start with KOKKOS_, variables not for consumation by application Makefiles start with KOKKOS_INTERNAL.

Functions start with kokkos_ (note lower case) so this ifhas function would become kokkos_ifhas.

Reply:
I'm confused by this comment.  Are you referring to `KOKKOS_CUDA_OPTIONS`?

Reply:
No I mean we need a bit more comment on what the function ifhas does.